src/create-timed-keyed-resource.test.js

```diff
@@ -0,0 +1,86 @@
+import React from "react";
+import { render, waitForElement } from "react-testing-library";
+import { identity } from "ramda";
+import useEntry from "./use-entry";
+import RemoteResourceBoundary from "./RemoteResourceBoundary";
+import createTimedKeyedResource from "./create-timed-keyed-resource";
+
+// ---------------------------
+// Mocks
+// ---------------------------
+
+const MockResourceConsumer = ({ resource, index }) => {
+  const [entry] = useEntry(resource, [index]);
+
+  return entry;
+};
+
+// ---------------------------
+// Tests
+// ---------------------------
+
+describe("createTimedKeyedResource", () => {
+  it("creates a resource", async () => {
+    const resource = createTimedKeyedResource(10, identity, () =>
+      Promise.resolve("resolved")
+    );
+
+    const { getByText } = render(
+      <RemoteResourceBoundary
+        fallback={<p>Loading...</p>}
+        renderError={() => <p>error</p>}
+      >
+        <MockResourceConsumer resource={resource} index={0} />
+      </RemoteResourceBoundary>
+    );
+
+    await waitForElement(() => getByText("Loading..."));
+    await waitForElement(() => getByText("resolved"));
+  });
+
+  it("loads new data after the specified timeout", async () => {
+    let count = 0;
+
+    const resource = createTimedKeyedResource(100, identity, () => {
+      count += 1;
+      return Promise.resolve(count);
+    });
+
+    const { getByText, rerender } = render(
+      <RemoteResourceBoundary
+        fallback={<p>Loading...</p>}
+        renderError={() => <p>error</p>}
+      >
+        <MockResourceConsumer resource={resource} index={0} />
+      </RemoteResourceBoundary>
+    );
+
+    await waitForElement(() => getByText("Loading..."));
+    await waitForElement(() => getByText("1"));
+
+    rerender(
+      <RemoteResourceBoundary
+        fallback={<p>Loading...</p>}
+        renderError={() => <p>error</p>}
+      >
+        <MockResourceConsumer resource={resource} index={0} />
+      </RemoteResourceBoundary>
+    );
+
+    await waitForElement(() => getByText("1"));
+
+    await delay(100);
+
+    rerender(
+      <RemoteResourceBoundary
+        fallback={<p>Loading...</p>}
+        renderError={() => <p>error</p>}
+      >
+        <MockResourceConsumer resource={resource} index={0} />
+      </RemoteResourceBoundary>
+    );
+
+    await waitForElement(() => getByText("Loading..."));
+    await waitForElement(() => getByText("2"));
+  });
+});
```

Comment:
Now that I'm thinking about it, I don't know that we want this kind of resource to work this way. I'm not sure if our test is incorrect, or if we have written the resource creator incorrectly. But I would expect this resource to not re-load the data until the component was being rendered for the first time. Let's say the user is filling out a form, and it's taking them a while, and at some point the component re-renders after the timeout. All of a sudden the `fallback` would be shown because it's re-fetching the data. And according to this test that would happen.

We should probably change our test, asserting that the loader doesn't get called until we do a re-mount, even if the data times out. We can use [`unmount`](https://testing-library.com/docs/react-testing-library/api#unmount) from `react-testing-library` to simulate this.